u2net_test.py

```diff
@@ -1,116 +1,129 @@
 
     predict = pred
     predict = predict.squeeze()
     predict_np = predict.cpu().data.numpy()
 
-    im = Image.fromarray(predict_np*255).convert('RGB')
+    im = Image.fromarray(predict_np * 255).convert("RGB")
     img_name = image_name.split("/")[-1]
     image = io.imread(image_name)
-    imo = im.resize((image.shape[1],image.shape[0]),resample=Image.BILINEAR)
+    imo = im.resize((image.shape[1], image.shape[0]), resample=Image.BILINEAR)
 
     pb_np = np.array(imo)
 
     aaa = img_name.split(".")
     bbb = aaa[0:-1]
     imidx = bbb[0]
-    for i in range(1,len(bbb)):
+    for i in range(1, len(bbb)):
         imidx = imidx + "." + bbb[i]
 
-    imo.save(d_dir+imidx+'.png')
+    imo.save(d_dir + imidx + ".png")
+
 
 def main():
 
     # --------- 1. get image path and name ---------
-    model_name='u2net'#u2netp
+    model_name = "u2net"  # u2netp
 
+    image_dir = "./test_data/test_images/"
+    prediction_dir = "./test_data/" + model_name + "_results/"
+    model_dir = "./saved_models/" + model_name + "/" + model_name + ".pth"
 
-    image_dir = './test_data/test_images/'
-    prediction_dir = './test_data/' + model_name + '_results/'
-    model_dir = './saved_models/'+ model_name + '/' + model_name + '.pth'
-
-    img_name_list = glob.glob(image_dir + '*')
+    img_name_list = glob.glob(image_dir + "*")
     print(img_name_list)
 
     # --------- 2. dataloader ---------
-    #1. dataloader
-    test_salobj_dataset = SalObjDataset(img_name_list = img_name_list,
-                                        lbl_name_list = [],
-                                        transform=transforms.Compose([RescaleT(320),
-                                                                      ToTensorLab(flag=0)])
-                                        )
-    test_salobj_dataloader = DataLoader(test_salobj_dataset,
-                                        batch_size=1,
-                                        shuffle=False,
-                                        num_workers=1)
+    # 1. dataloader
+    test_salobj_dataset = SalObjDataset(
+        img_name_list=img_name_list,
+        lbl_name_list=[],
+        transform=transforms.Compose([RescaleT(320), ToTensorLab(flag=0)]),
+    )
+    test_salobj_dataloader = DataLoader(
+        test_salobj_dataset, batch_size=1, shuffle=False, num_workers=1
+    )
 
     # --------- 3. model define ---------
-    if(model_name=='u2net'):
+    if model_name == "u2net":
         print("...load U2NET---173.6 MB")
-        net = U2NET(3,1)
-    elif(model_name=='u2netp'):
+        net = U2NET(3, 1)
+    elif model_name == "u2netp":
         print("...load U2NEP---4.7 MB")
-        net = U2NETP(3,1)
-    net.load_state_dict(torch.load(model_dir))
+        net = U2NETP(3, 1)
+
     if torch.cuda.is_available():
+        net.load_state_dict(torch.load(model_dir))
         net.cuda()
+    else:
+        net.load_state_dict(torch.load(model_dir, map_location=torch.device("cpu")))
+
```

Comment:
If you try to load_state_dict on CPU without mapping location to CPU, you will have a RuntimeError
`RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.`

Reply:
I have tried this and it gives me `TypeError: 'torch.Device' object is not callable`

Reply:
@MatiasConTilde this should also work with both torch>=0.4 with map_location="cpu"